app/models/base_facility.rb

```diff
@@ -35,6 +67,48 @@ def pull_source_data
       max_record_count = metadata['maxRecordCount']
       Facilities::SharedClient.new.get_all_facilities(*FACILITY_SORT_FIELDS[name], max_record_count).map(&method(:new))
     end
+
+    def find_facility_by_id(id)
+      type, unique_id = id.split('_')
+      return nil unless type && unique_id
+      "Facilities::#{type.upcase}Facility".constantize.find_by(unique_id: unique_id)
+    end
+
+    def query(params)
+      bbox_num = params[:bbox].map { |x| Float(x) }
+      data = build_result_set(bbox_num, params[:type], params[:services])
+      Common::Collection.new(::VAFacility, data: data.sort_by(&(dist_from_center bbox_num)))
```

Comment:
Same comment here, returning a Common::Collection is not going to be chainable as an ActiveRecord::Relation.

I know you needed Common::Collection to ensure the metadata for pagination links gets populated correctly, but longer term we should revisit this and come up with a better solution. There should be a concern we could mixin instead.